editor/src/clj/editor/future.clj

```diff
@@ -29,14 +29,36 @@
   ^CompletableFuture [x]
   (CompletableFuture/completedFuture x))
 
-(defmacro supply-async [& body]
+(defmacro compute
+  "Asynchronously perform a CPU-intensive operation (see also: io)"
+  [& body]
   `(let [bindings# (get-thread-bindings)]
      (CompletableFuture/supplyAsync
        (reify Supplier
          (get [~'_]
            (with-bindings bindings#
              ~@body))))))
 
+(def io-executor
+  (let [counter (atom 0)]
+    (Executors/newCachedThreadPool
+      (reify ThreadFactory
+        (newThread [_ r]
+          (doto (Thread. r)
```

Comment:
Reflection warning?

Reply:
Hmm no, I don't see any reflection warnings, and there is `(set! *warn-on-reflection* true)` at the top. This is probably because `r` is type-hinted in the `reify` since `newThread` defines a `Runnable` arg.